zaza/openstack/charm_tests/vault/tests.py

```diff
@@ -153,7 +153,8 @@ def test_csr(self):
             allowed_domains='openstack.local')
 
         test_config = lifecycle_utils.get_charm_config()
-        del test_config['target_deploy_status']['vault']
+        test_config['target_deploy_status']['vault'][
+            'workload-status-message'] = 'New version of vault installed'
```

Comment:
Two things:
1. this message isn't guaranteed to happen, is it? 
2. this message isn't going to happen with older versions of the charm (we want these tests to remain backwards compatible with older released versions of the charms)

Not sure how to proceed here.

Reply:
Hi

The test is backward compatible. The new status message will be added to pre-defined list of approved status messages [1].
I also verified by running this PR against charm vault master branch.

[1] https://github.com/openstack-charmers/zaza/blob/master/zaza/model.py#L1051

Reply:
Thank you for the patch, however this is not how the list of approved status messages is used.

The authoritative source of information for expected deploy-time status messages is the `tests/tests.yaml` in the charm repository (when they diverge from the default 'active', 'Unit is ready' pattern).

During the course of executing functional tests the test code may need to adjust the expected status message to match what the functional test is doing.

In this case I don't see you adding a snap upgrade test, so the change is redundant.

Reply:
Thanks for the review comments.

The status message that gets changed in this patch is specific to execution of functional test.
Currently the vault versions are broken due to bugs in vault. Vault installs version 1.5.3 but reports as 1.5.0.
This is fixed in vault 1.5.4 but awaiting snap release for vault 1.5.4 [1].

Until vault snap  for 1.5.4 is released, the vault tests will be failed for [2] without this PR. 
Do you suggest to wait until [1] is released so that we dont need this PR (as you mentioned this change is redundant and only solving to overcome the temporary vault versions issue)

[1] https://github.com/canonical/snap-vault/issues/26
[2] https://review.opendev.org/#/c/754267/

Reply:
Thanks for the explanation and I think this whole explanation should be present as a note in the code. Also please provide this information upfront when creating a pull request next time as we have no chance to guess it as reviewers. This PR gave us the illusion that it was solving https://review.opendev.org/#/c/754267/ where in reality this is a workaround for canonical/snap-vault#26 . Thanks!

And also I agree with @fnordahl I still think this should be in `tests/tests.yaml` instead.